domainservice/src/main/kotlin/systems/ajax/malov/stockanalyzer/mapper/proto/StockPriceMapper.kt

```diff
@@ -1,28 +1,19 @@
 package systems.ajax.malov.stockanalyzer.mapper.proto
 
 import com.google.protobuf.Timestamp
-import com.squareup.wire.Instant
 import systems.ajax.malov.internalapi.output.pubsub.stock.stock_price.proto.StockPrice
 import systems.ajax.malov.stockanalyzer.entity.MongoStockRecord
 import systems.ajax.malov.stockanalyzer.mapper.proto.BigDecimalProtoMapper.convertToBigDecimalProto
+import systems.ajax.malov.stockanalyzer.mapper.proto.TimestampProtoMapper.toTimestampProto
 
 object StockPriceMapper {
     fun MongoStockRecord.toStockPrice(): StockPrice {
         return StockPrice.newBuilder()
             .setPrice(convertToBigDecimalProto(currentPrice))
             .setStockSymbolName(symbol)
             .setTimestamp(
-                dateOfRetrieval?.let {
-                    toTimestampProto(it)
-                } ?: Timestamp.getDefaultInstance()
+                dateOfRetrieval?.toTimestampProto() ?: Timestamp.getDefaultInstance()
```

Comment:
Timestamp.getDefaultInstance() and not set timestamp are different semantically

here I'd prefer that null instant is mapped as 'not set' timestamp